From 849ae4fe498f2435111939a78dab8438cb54e385 Mon Sep 17 00:00:00 2001
From: Fernando Luiz Cola <fernando.luiz@pantacor.com>
Date: Thu, 20 Aug 2026 18:27:01 -0300
Subject: [PATCH] configs: rockchip-common: run distro_bootcmd directly, not
 Android boot.img

RKIMG_BOOTCOMMAND (compiled into the default "bootcmd" environment via
CONFIG_EXTRA_ENV_SETTINGS/ROCKCHIP_DEVICE_SETTINGS) is a plain C macro,
independent of the Kconfig CONFIG_BOOTCOMMAND symbol -- so
meta-pantavisor's pv.distroboot.cfg fragment (CONFIG_BOOTCOMMAND="run
distro_bootcmd") has no effect on it. Since meta-rockchip's
generic-gptdisk.wks.in always populates the "boot" partition with a
valid Android-format boot.img, "boot_android" always succeeds and
boots it directly with vendor-Android bootargs -- no root=, no
initrd -- which panics: "VFS: Unable to mount root fs".

Signed-off-by: Fernando Luiz Cola <fernando.luiz@pantacor.com>
---
 include/configs/rockchip-common.h | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/include/configs/rockchip-common.h b/include/configs/rockchip-common.h
index 002dcd6..5e36b43 100644
--- a/include/configs/rockchip-common.h
+++ b/include/configs/rockchip-common.h
@@ -169,10 +169,13 @@
 #define RKIMG_BOOTCOMMAND			\
 	"boot_fit;"
 #else
+/*
+ * Pantavisor: go straight to distro_bootcmd. The vendor Android/FIT/rkp
+ * paths boot the "boot" partition's Android boot.img (which
+ * generic-gptdisk.wks.in populates) with bootargs that have no
+ * root=/initrd, panicking with "VFS: Unable to mount root fs".
+ */
 #define RKIMG_BOOTCOMMAND			\
-	"boot_android ${devtype} ${devnum};"	\
-	"boot_fit;"				\
-	"bootrkp;"				\
 	"run distro_bootcmd;"
 #endif
 
-- 
2.43.0

